### DISCUSSION TITLE
argo-workflows/quick-start/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#8629] [General]
# argo-workflows/quick-start/

https://argoproj.github.io/argo-workflows/quick-start/

**Comment by VladoPortos** [comment]
Guys, you can't just end with `This will serve the UI on https://localhost:2746` because the moment you open the UI you get hit with log in page. How to log in ? Deploying from helm chart, I can see a secret created called, 
`argo-workflow-argo-workflows-server-token-jvvnj` is that a token it needs ?

**Reply by VladoPortos** [reply]
Ok you need to decode the token from the created `argo-workflow-argo-workflows-server-token-xxx` but also needs to add `Barer <token>` and log in with that :) would be nice to mention it here.

**Reply by omBratteng** [reply]
If you start with `argo server --auth-mode=server` you won't see the login page, which perhaps is a better starting point.

**Reply by VladoPortos** [reply]
@omBratteng any idea what parameter it is in Helm chart ? I could not find anything like that so far.

**Reply by omBratteng** [reply]
`server.extraArgs: [--auth-mode=server]`

**Reply by VladoPortos** [reply]
@omBratteng Thanks, I think I got it working. The `server.extraArgs` does not show in Argo CD as option in values. So I have added it to manifest to custom helm values as:

```yaml
      - name: server.extraArgs
        value: '{--auth-mode=server}'
```

That's seems to help.

**Reply by seek4** [reply]
You can try this command:
```sh
argo server --auth-mode server
```
It works for me.

**Reply by tim-sendible** [reply]
I have modified this page to include steps on how to enable server authentication.

**Reply by jrbe228** [reply]
I had this error when using `--auth-mode=server`: "Request has been terminated Possible causes: the network is offline, Origin is not allowed by Access-Control-Allow-Origin, the page is being unloaded, etc." 

It was fixed by adding the following flag: 
```sh
argo -n argo server --access-control-allow-origin=* --auth-mode=server
```

**Reply by mwoodpatrick** [reply]
I hit this issue and it fixed the issue for me also. I tried using this in the command line suggested in the getting started:

```yaml
kubectl patch deployment  argo-server  --namespace ${ARGO_NS} --type='json' -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/args", "value": [ "server", "--auth-mode=server", "--access-control-allow-origin=*" ]}]'
```

Sometimes it appears to work and sometimes when I access the UI I get:

"Request has been terminated Possible causes: the network is offline, Origin is not allowed by Access-Control-Allow-Origin, the page is being unloaded, etc."

**Reply by worldsayshi** [reply]
I'm using argo-workflows:0.41.2 from https://argoproj.github.io/argo-helm. When trying `argo server --auth-mode=server` I get `configmaps "workflow-controller-configmap" not found`. There is one configmap called `argo-workflows-workflow-controller-configmap`. Seems the helm chart is prepending the releaseName to the configmap and the `argo` cli doesn't expect that.

I could remedy that by running:
```
argo -n argo server --access-control-allow-origin=* --auth-mode=server --configmap argo-workflows-workflow-controller-configmap
```

I can now get into the GUI at https://localhost:2746

**Reply by agilgur5** [reply]
@worldsayshi the quick-start is meant to be run via the quick-start manifests, as it says on this very page.

So that makes sense that if you install via `argo-helm`, your config will differ, as you didn't follow the installation directions.

**Reply by worldsayshi** [reply]
I see. I thought the installation manifest was the same. I switched to the helm chart to try to find a way to get past the login screen. I guess with the quick start manifest i wouldn't have had that issue to begin with..

**Comment by yaoxinwang-git** [comment]
If you want to try out Argo Workflows and don't want to set up a Kubernetes cluster, the community is working on a replacement for the old Katacoda course since Katacoda was shut down. Please give a thumbs up or comment on this issue with your support and feedback.

**Reply by terrytangyuan** [reply]
Which issue are you referring to? This is a discussion thread for quick start guide.

**Reply by rish8089dunzo** [reply]
@yaoxinwang-git
where can I find replacement for the old Katacoda course?

**Reply by tico24** [reply]
@rish8089dunzo It doesn't exist yet. We are hoping to have something out in the next week or so.

**Reply by caelan-io** [reply]
@rish8089dunzo @yaoxinwang-git Hello, You can find the new course here on Killercoda, a Katacoda replacement: https://killercoda.com/pipekit/course/argo-workflows

@tico24 and I have posted a new version of the Intro to Argo Workflows course, compatible with v3.4.2. We're looking for more user feedback before improving on and adding to the new course. 

For more info, see issue #8899

**Comment by yaoxinwang-git** [comment]
when i want to use UI, I curl  service argo-server clusterIP:port  ,tell me Empty reply from server,  please tell how to visit UI

**Reply by yaoxinwang-git** [reply]
edit service type nodeport  then use https://ip:nodeport to visit ui

**Comment by oschrenk** [comment]
I strongly discourage from using the manifest mentioned in the beginning of the document. It doesn't work. You are better off using a quickstart manifest attached to a release eg 

```
kubectl apply -n argo -f https://github.com/argoproj/argo-workflows/releases/download/v3.3.9/quick-start-postgres.yaml
```

I ran into 

```
W0810 13:54:18.440110       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.3/tools/cache/reflector.go:167: failed to list *v1alpha1.WorkflowArtifactGCTask: the server could not find the requested resource (get workflowartifactgctasks.argoproj.io)
```

where artifacts didn't work.  See also https://cloud-native.slack.com/archives/C01QW9QSSSK/p1660203791944519

Others ran into issues with the cron flows.

**Reply by tim-sendible** [reply]
I have modified this page to point users to the correct manifests now.

**Comment by RiyajKazieomccain** [comment]
Hi Team,
 
I have deployed the argo-workflow using helm, whenever i try to submit the workflow getting below error. Does anyone know the solution for the same.

```
Forbidden: workflows.argoproj.io is forbidden: User "system:serviceaccount:argo-workflows:default" cannot create resource "workflows" in API group "argoproj.io" in the namespace "default"
```

**Reply by Souheil-Yazji** [reply]
Check the ClusterRoleBinding mentioned in the error and associated Cluster role for the service account "argo-workflows". Make sure that it has "CREATE" permissions for the "workflows" resource in API group "argoproj.io".

Check out https://github.com/argoproj/argo-helm/tree/main/charts/argo-workflows#pre-requisites which might address your issue.

**Comment by zhouduo0** [comment]
how to create argo namespace

**Reply by Souheil-Yazji** [reply]
`kubectl create namespace argo` is provided in the docs above

**Comment by Alig1493** [comment]
Shouldn't the port forwarding be done for the service instead of the deployments?

**Reply by tico24** [reply]
From a technical standpoint it doesn't matter. PFing a service just ends up pointing at the pod anyway. From an ease of use perspective, it might be a little easier I guess as the name doesn't change. However, in this case we PF the deployment anyway, which works around the name-change issue.

PRs are welcome.

**Comment by Hanliber** [comment]
Every time I submit the same workflow，argo always create some new pods to run this workflow, are there some other ways to reuse the pods which this workflow used to create?

**Reply by fsedano** [reply]
The idea is each workflow starts from a clean and well known state, so pod reuse does not look the right thing to do. Can you elaborate on the use case for this?

**Reply by Hanliber** [reply]
Can I define each task in the workflow as a type of deployment? In this way, every time the workflow is run, each pod will not be in the completed state.
